drivers/i2c/i2c_mchp_xec.c

```diff
@@ -69,6 +72,55 @@ static const struct xec_speed_cfg xec_cfg_params[] = {
 	},
 };
 
+static void i2c_xec_reconfig(uint32_t ba)
```

Comment:
Here is the sequence from working MCHP code. This code sequence is called to enable a controller and if a bus reset is required.
1. Reset controller: Set bit[9]=1 in I2C Configuration register (offset 0x28). Delay > 1 16MHz clock cycle, Set bit[9]=0 in I2C Configuration register.  Your latest code appears to be doing this.
2. Write 0x80 to I2C Control register (offset 0x00).
3. Write I2C Configuration register (not read-modify-write).   Set bit[14]=1 to disable I2C general call, set b[10]=1 to enable the I2C block, set bit[8]=1 to enable I2C filtering unless you have external HW filters on your board, and set bits[3:0] to the default port.  Please use a write not read-modify-write for this step.
4. Program timing registers: Bus clock(offset 0x2C), data timing(offset 0x40), timeout scaling(offset 0x44), data timing2(offset 0x18).
5. Program Own Address(offset 0x04) value
6. Write 0xC1 to I2C Control register (offset 0x00).

Reply:
Followed this.